### PR TITLE
Remove usages of deprecated created-by label (fixes #842)

### LINKED ISSUE #842: Refactor our usage of labels
Alex Dutra wrote a [fantastic analysis](https://docs.google.com/document/d/1bXkEBtxtQXiyr5d3I4RzuNpw0LMFHO3kozUrMLM5xhk/edit#heading=h.pid5o0551qj9) of our usage of labels, along with some strong suggestions on how to refactor them.
We have a lot of implicit rules and some confusing sets of labels (managed-by vs part of), along with inconsistent ways of applying sets of labels (SetManagedBy() exists but not SetPartOf() for example).

We'd need to review Alex's recommendations and do the suggested refactorings.

- [x] #859
- [x] #860
- [ ] #861

### PR DESCRIPTION
**Pull request by olim7t**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #842

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: pkg/reaper/labels.go [hunk 1]
```diff
@@ -14,18 +16,14 @@ var commonLabels = map[string]string{
...
+	return utils.MergeMap(
+		commonLabels,
+		k8ssandralabels.CleanedUpByLabels(client.ObjectKey{Namespace: kc.Namespace, Name: kc.Name}))
 }
 
 func getConstantLabels(r *reaperapi.Reaper) map[string]string {
 	labels := map[string]string{
-		reaperapi.ReaperLabel:       r.Name,
-		k8ssandraapi.CreatedByLabel: k8ssandraapi.CreatedByLabelValueReaperController,
+		reaperapi.ReaperLabel: r.Name,
```
Comment (olim7t) [inline pkg/reaper/labels.go:26]: This is used in the service's selector, I've checked that removing a label from it is not a breaking change.

File: pkg/telemetry/prom_stargate_servicemonitor.go [hunk 2]
```diff
@@ -138,8 +137,7 @@ func (cfg PrometheusResourcer) NewStargateServiceMonitor() (promapi.ServiceMonit
 		Spec: promapi.ServiceMonitorSpec{
 			Selector: metav1.LabelSelector{
 				MatchLabels: map[string]string{
-					k8ssandraapi.CreatedByLabel: k8ssandraapi.CreatedByLabelValueStargateController,
-					stargateapi.StargateLabel:   cfg.MonitoringTargetName,
+					stargateapi.StargateLabel: cfg.MonitoringTargetName,
```
Comment (olim7t) [inline pkg/telemetry/prom_stargate_servicemonitor.go:140]: ServiceMonitor selectors are also mutable.

File: pkg/cassandra/datacenter.go [hunk 3]
```diff
@@ -153,14 +153,11 @@ func NewDatacenter(klusterKey types.NamespacedName, template *DatacenterConfig)
...
-				api.ComponentLabel:                 api.ComponentLabelValueCassandra,
-				api.CreatedByLabel:                 api.CreatedByLabelValueK8ssandraClusterController,
-				api.K8ssandraClusterNameLabel:      klusterKey.Name,
-				api.K8ssandraClusterNamespaceLabel: klusterKey.Namespace,
-			},
+			Labels: utils.MergeMap(map[string]string{
+				api.NameLabel:      api.NameLabelValue,
+				api.PartOfLabel:    api.PartOfLabelValue,
+				api.ComponentLabel: api.ComponentLabelValueCassandra,
+			}, labels.CleanedUpByLabels(klusterKey)),
```
Comment (adejanovski) [inline pkg/cassandra/datacenter.go:160]: Nice!

**Review by adejanovski — APPROVED** [review]
Looks great!
You'll have to rebase and fix the changelog conflict before this can be merged.

**Pull request merged**